src/shared/install.c

```diff
@@ -2482,14 +2507,20 @@ int unit_file_link(
                 return -ENXIO;
 
         STRV_FOREACH(file, files) {
-                _cleanup_free_ char *full = NULL;
+                _cleanup_free_ char *full = NULL, *fname = NULL;
                 struct stat st;
                 char *fn;
 
                 if (!path_is_absolute(*file))
                         return install_changes_add(changes, n_changes, -EINVAL, *file, NULL);
 
-                fn = basename(*file);
```

Comment:
Please drop this part of the change. I have a bigger rework at https://github.com/systemd/systemd/pull/31588/commits/dd358987123a986a79b4702a864795d20abb84d0 that includes this.